transitions/diagrams.py

```diff
@@ -0,0 +1,63 @@
+import pygraphviz as pgv
+
+
+class Diagram(object):
+    def __init__(self, machine):
+        self.machine = machine
+
+
+class AGraph(Diagram):
```

Comment:
Maybe add get_graph to Diagram as an @abstractmethod to make sure it's implemented by any future graphing interfaces.